test/functional/rpc_packages.py

```diff
@@ -369,6 +369,14 @@ def test_submit_child_with_parents(self, num_parents, partial_submit):
     def test_submitpackage(self):
         node = self.nodes[0]
 
+        self.log.info("Submitpackage only allows valid hex inputs")
+        valid_tx_list = self.wallet.create_self_transfer_chain(chain_length=2)
+        hex_list = [valid_tx_list[0]["hex"][:-1] + 'X', valid_tx_list[1]["hex"]]
+        txid_list = [valid_tx_list[0]["txid"], valid_tx_list[1]["txid"]]
+        assert_raises_rpc_error(-22, "TX decode failed:", node.submitpackage, hex_list)
```

Comment:
Induced failure with:
```diff
-        hex_list = [valid_tx_list[0]["hex"][:-1] + 'X', valid_tx_list[1]["hex"]]
+        hex_list = [valid_tx_list[0]["hex"], valid_tx_list[1]["hex"]]
```
Also verified that the `not in node.getrawmempool()` asserts also fail, e.g.
```diff
         self.log.info("Submitpackage only allows valid hex inputs")
         valid_tx_list = self.wallet.create_self_transfer_chain(chain_length=2)
-        hex_list = [valid_tx_list[0]["hex"][:-1] + 'X', valid_tx_list[1]["hex"]]
+        hex_list = [valid_tx_list[0]["hex"], valid_tx_list[1]["hex"]]
         txid_list = [valid_tx_list[0]["txid"], valid_tx_list[1]["txid"]]
-        assert_raises_rpc_error(-22, "TX decode failed:", node.submitpackage, hex_list)
-        assert txid_list[0] not in node.getrawmempool()
+        node.submitpackage(hex_list)
+        #assert_raises_rpc_error(-22, "TX decode failed:", node.submitpackage, hex_list)
+        #assert txid_list[0] not in node.getrawmempool()
         assert txid_list[1] not in node.getrawmempool()
```